src/stan_math_backend/Transform_Mir.ml

```diff
@@ -161,56 +161,97 @@ let data_read smeta (decl_id, st) =
           ; Stmt.Helpers.for_scalar_inv st bodyfn decl_var smeta ]
         |> swrap ]
 
-let rec base_ut_to_string = function
-  | UnsizedType.UMatrix -> "matrix"
-  | UVector -> "vector"
-  | URowVector -> "row_vector"
-  | UReal -> "scalar"
-  | UInt -> "integer"
-  | UArray t -> base_ut_to_string t
-  | t ->
-      raise_s
-        [%message "Another place where it's weird to get " (t : UnsizedType.t)]
+type constrainaction = Check | Constrain | Unconstrain [@@deriving sexp]
+
+let check_constraint_to_string t (c : constrainaction) =
```

Comment:
I left it because we're leaving unconstrains for another PR, but I can still probably clean it up

Reply:
I think this'll actually be a bit annoying to clean up since Ast_to_mir shouldn't depend on Transform_mir. I'd prefer to clean this up with everything else in the unconstrain PR